pkgs/servers/http/apache-httpd/2.4.nix

```diff
@@ -38,11 +38,16 @@ stdenv.mkDerivation rec {
   # Required for ‘pthread_cancel’.
   NIX_LDFLAGS = lib.optionalString (!stdenv.isDarwin) "-lgcc_s";
 
+  # TODO: the configure script fails to find a working pcre2-config without this
+  preConfigure = ''
+    export PATH=${lib.getDev pcre2}/bin:$PATH
+  '';
```

Comment:
This is not ideal and if someone has time to check the upstream changes that resulted in that breakage I'm all for it.

Reply:
The `which` command is missing while trying to determine the pcre2-config location when setting it via [`--with-pcre`](https://github.com/apache/httpd/blob/2.4.53/configure.in#L215-L225). 

Adding `which` to the inputs and setting `--with-prce` to `${pcre2.dev}/bin/pcre2-config` fixes the build without the workaround.

```
...
apache-httpd> checking for gcc option to accept ISO C99... none needed
apache-httpd> /nix/store/3vygpb5d5n7gq5f7k30zvpv2sgnw5r1v-pcre2-10.39-dev/bin/pcre2-config
apache-httpd> checking for -pcre2-config... no
apache-httpd> checking for -pcre-config... no
apache-httpd> checking for pcre2-config... no
apache-httpd> checking for pcre-config... no
apache-httpd> configure: Using external PCRE library from /nix/store/3vygpb5d5n7gq5f7k30zvpv2sgnw5r1v-pcre2-10.39-dev/bin/pcre2-config
apache-httpd>   setting PCRE_INCLUDES to "-I/nix/store/3vygpb5d5n7gq5f7k30zvpv2sgnw5r1v-pcre2-10.39-dev/include"
apache-httpd>   setting PCRE_LIBS to "-L/nix/store/4vxdmr77q3kzf223vzyxbr82mj82203y-pcre2-10.39/lib -lpcre2-8"
apache-httpd> configure:
apache-httpd> configure: Configuring Apache httpd...
...
```

Reply:
In 2.4.52 the [check](https://github.com/apache/httpd/blob/2.4.52/configure.in#L215-L223) worked without `which` and appended `/bin/pcre-config`to the `--with-pcre` path.

Reply:
Thanks for looking into this.